managed-pointer/cpp-20/D0567R1.md

```diff
@@ -5,222 +5,384 @@
+
+The major distinction is whether or not the `executor_type` of a `managed_ptr` is a *host* which or a *remote*. If the `executor_type` is a *host* then it supports host based capabilities; being able to perform allocation and sycnhronisation operations, if it is a *remote* it does not. Whether the `executor_type` of a `managed_ptr` is a *host* can be queried with the static constexpr bool `is_host_based`.
+
+> For the purposes of this paper the capabilities of host and remote execution contexts simply state that allocation and sycnhronisation operations can only be performed by host execution contexts. In reality the capabilities of execution contexts will be far more complex and will require an depth interface for querying said capabilities. However this problem of language capabilities on execution contexts particularly for heterogeneous architectures is a much larger problem for C\+\+ that goes beyond the scope of this paper and indeed the executors proposal as well.
 
-*Figure 2: Extensions to unified interface for execution*
+#### 3.3.3. Memory accessibiity
 
-As described previously there are three ways in which a synchronisation operation can be triggered, and these are separated into explicit and implicit.
+During the lieftime of a `managed_ptr` it's managed memory allocation can exist in the memory region(s) associated with any number of *execution contexts*, but may only be accessible in one of these memory regions at any given time. The *execution context* who's memory region is currently accessible is said to be the accessible *execution context*.
 
-The first is by calling one of the member functions or global customisation point functions described above. This will trigger a synchronisation operation to the memory region of the **execution context** associated with the **executor**. This method is useful as it allows finer control over the synchronisation operations by chaining continuations. This is implicit.
+> This limited may be relaxed in a future version of the paper when a `managed_ptr` can be concurrently accessed in more than one memory region. In this case there could be more than one accessible memory region. This is further described in section 5.1.
 
-The second is by calling the ``get()`` member function on the ``managed_ptr`` itself as this will implicitly trigger a call to ``get()`` on the current accessible **execution context** if the host CPU is not currently accessible. This method is useful when you want to synchronise data back to the host CPU in order to use the resulting data in regular sequential code. This is explicit.
+The `managed_ptr` must maintain a reference to the current accessible *execution context*.
```

Comment:
a reference or a copy? Can the execution context instance be deleted or has to be alive? There is a fundamental difference in the sense that if an execution context destructor has some expected behaviour (e.g synchronization), which will be triggered then by a chain of destructors from the managed_ptr if a copy is stored, but not if there is a reference.